pkg/cmd/kafka/create/create.go

```diff
@@ -579,48 +574,84 @@ func promptKafkaPayload(opts *options, constants *remote.DynamicServiceConstants
 		}
 	}
 
+	// if billing model is marketplace, prompt for marketplace provider and account id
 	if answers.BillingModel == accountmgmtutil.QuotaMarketplaceType {
-		validMarketPlaces := FetchValidMarketplaces(orgQuota.MarketplaceQuotas, answers.CloudProvider)
-		if len(validMarketPlaces) == 1 {
-			answers.Marketplace = validMarketPlaces[0]
-		} else {
-			marketplacePrompt := &survey.Select{
-				Message: f.Localizer.MustLocalize("kafka.create.input.marketplace.message"),
-				Options: validMarketPlaces,
-			}
-			err = survey.AskOne(marketplacePrompt, &answers.Marketplace)
-			if err != nil {
-				return nil, err
-			}
+		answers, err = marketplaceQuotaPrompt(orgQuota, answers, f, err)
+		if err != nil {
+			return nil, err
 		}
+	}
 
-		if len(validMarketPlaces) > 0 {
+	var sizes []kafkamgmtclient.SupportedKafkaSize
+	payload := &kafkamgmtclient.KafkaRequestPayload{}
 
-			validMarketplaceAcctIDs := FetchValidMarketplaceAccounts(orgQuota.MarketplaceQuotas, answers.Marketplace)
+	if opts.useEnterpriseCluster {
+		/*
+			if using dedicated cluster option then get the supported sizes for this cluster
+			based on the sizes it supports the user to select the one they want to use
+			then check if the size uses <= the remaining streaming units it takes up
+			on the cluster
+		*/
+		cluster, err := getClusterDetails(opts.f, opts.selectedCluster.GetId())
+		if err != nil {
+			return nil, err
+		}
 
-			if len(validMarketplaceAcctIDs) == 1 {
-				answers.MarketplaceAcctID = validMarketplaceAcctIDs[0]
-			} else {
-				marketplaceAccountPrompt := &survey.Select{
-					Message: f.Localizer.MustLocalize("kafka.create.input.marketplaceAccountID.message"),
-					Options: validMarketplaceAcctIDs,
-				}
-				err = survey.AskOne(marketplaceAccountPrompt, &answers.MarketplaceAcctID)
-				if err != nil {
-					return nil, err
-				}
-			}
+		supportedInstanceTypes := cluster.GetSupportedInstanceTypes()
+		supportedKafkaTypes := supportedInstanceTypes.GetInstanceTypes()
+
+		// right now enterprise clusters only support one type of kafka, so we are hard
+		// checking this, this assumption may change in the future
+		if len(supportedKafkaTypes) != 1 {
+			return nil, fmt.Errorf("expected one supported instance kafka type")
 		}
-	}
 
-	marketplaceInfo := accountmgmtutil.MarketplaceInfo{
-		BillingModel:   answers.BillingModel,
-		Provider:       answers.Marketplace,
-		CloudAccountID: answers.MarketplaceAcctID,
-	}
+		sizes = supportedKafkaTypes[0].Sizes
+
+		index, err := promptUserForSizes(opts.f, &sizes)
```

Comment:
no need to prompt for size if `opts.size` the flag has been set

Reply:
Will do

Reply:
So I was adding this and discovered that we didn't know exactly how the command handles flags. From what I can see when adding a flag like `size` or `name` the command won't just skip prompts in interactive mode, it skips interactive mode all together. Meaning there is no need t handle it in this case as above ^^